geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterStore.scala

```diff
@@ -0,0 +1,405 @@
+import org.locationtech.geomesa.accumulo.util.{SelfClosingBatchScanner, SelfClosingScanner}
+import org.locationtech.geomesa.raster._
+import org.locationtech.geomesa.raster.index.RasterIndexSchema
+import org.locationtech.geomesa.raster.util.RasterUtils
+import org.locationtech.geomesa.security.AuthorizationsProvider
+import org.locationtech.geomesa.utils.geohash.BoundingBox
+import org.locationtech.geomesa.utils.stats.{MethodProfiling, NoOpTimings, Timings, TimingsImpl}
+
+import scala.collection.JavaConversions._
+
+trait RasterOperations {
+  def getTable(): String
+  def ensureBoundsTableExists(): Unit
+  def createTableStructure(): Unit
+  def getAuths(): Authorizations
+  def getVisibility(): String
+  def getConnector(): Connector
+  def getRasters(rasterQuery: RasterQuery)(implicit timings: Timings): Iterator[Raster]
+  def getQueryRecords(numRecords: Int): Iterator[String]
+  def putRaster(raster: Raster): Unit
+  def getBounds(): BoundingBox
+  def getAvailableResolutions(): Seq[Double]
+  def getAvailableGeoHashLengths(): Set[Int]
+  def getAvailabilityMap(): ImmutableSetMultimap[Double, Int]
+  def getGridRange(): GridEnvelope2D
+  def getMosaicedRaster(query: RasterQuery, params: GeoMesaCoverageQueryParams): BufferedImage
+  def deleteRasterTable(): Unit
+}
+
+class AccumuloRasterStore(val connector: Connector,
+                  val tableName: String,
+                  val authorizationsProvider: AuthorizationsProvider,
+                  val writeVisibilities: String,
+                  shardsConfig: Option[Int] = None,
+                  writeMemoryConfig: Option[String] = None,
+                  writeThreadsConfig: Option[Int] = None,
+                  queryThreadsConfig: Option[Int] = None,
+                  collectStats: Boolean = false) extends RasterOperations with MethodProfiling with StatWriter with Logging {
+  //By default having at least as many shards as tservers provides optimal parallelism in queries
+  val shards = shardsConfig.getOrElse(connector.instanceOperations().getTabletServers.size())
+  val writeMemory = writeMemoryConfig.getOrElse("10000").toLong
+  val writeThreads = writeThreadsConfig.getOrElse(10)
+  val bwConfig: BatchWriterConfig =
+    new BatchWriterConfig().setMaxMemory(writeMemory).setMaxWriteThreads(writeThreads)
+  val numQThreads = queryThreadsConfig.getOrElse(20)
+
+  // TODO: WCS: GEOMESA-585 Add ability to use arbitrary schemas
+  val schema = RasterIndexSchema()
+  lazy val queryPlanner: AccumuloRasterQueryPlanner = new AccumuloRasterQueryPlanner(schema)
+
+  private val tableOps    = connector.tableOperations()
+  private val securityOps = connector.securityOperations
+  private val profileTable  = s"${tableName}_queries"
+  private def getBoundsRowID = tableName + "_bounds"
+  
+  //TODO: WCS: this needs to be implemented .. or  maybe not
+  //lazy val aRasterReader = new AccumuloRasterReader(tableName)
```

Comment:
Looks like we should establish whether this line is needed (and use it) or not (and remove it).